lib/Connection.php

```diff
@@ -482,110 +472,125 @@ private function validateConfiguration() {
 			return false;
 		}
-		if (!$this->ldapConnectionRes) {
-			if (!$this->getLDAP()->areLDAPFunctionsAvailable()) {
-				$phpLDAPinstalled = false;
-				Util::writeLog('user_ldap',
-									'function ldap_connect is not available. Make '.
-									'sure that the PHP ldap module is installed.',
-									Util::ERROR);
-
-				return false;
-			}
-			if ($this->configuration->turnOffCertCheck) {
-				if (\putenv('LDAPTLS_REQCERT=never')) {
-					Util::writeLog('user_ldap',
-						'Turned off SSL certificate validation successfully.',
-						Util::DEBUG);
-				} else {
-					Util::writeLog('user_ldap',
-										'Could not turn off SSL certificate validation.',
-										Util::WARN);
-				}
+		if ($this->configuration->turnOffCertCheck) {
+			if (\putenv('LDAPTLS_REQCERT=never')) {
+				Util::writeLog(
+					'user_ldap',
+					'Turned off SSL certificate validation successfully.',
+					Util::DEBUG
+				);
+			} else {
+				Util::writeLog(
+					'user_ldap',
+					'Could not turn off SSL certificate validation.',
+					Util::WARN
+				);
 			}
+		}
 
-			try {
-				// skip contacting main server after failed connection attempt
-				// until cache TTL is reached
-				if (\trim($this->configuration->ldapBackupHost) === ""
-					|| (!$this->configuration->ldapOverrideMainServer
-					&& !$this->getFromCache('overrideMainServer'))
-				) {
-					$this->doConnect(
-							$this->configuration->ldapHost,
-							$this->configuration->ldapPort);
-					if (@$this->ldap->bind(
-							$this->ldapConnectionRes,
-							$this->configuration->ldapAgentName,
-							$this->configuration->ldapAgentPassword)
-					) {
-						return true;
-					}
-					Util::writeLog('user_ldap',
-						'Bind failed: ' . $this->getLDAP()->errno($this->ldapConnectionRes) . ': ' . $this->getLDAP()->error($this->ldapConnectionRes),
-						Util::DEBUG); // log only in debug mod because this is triggered by wrong passwords
-					throw new BindFailedException();
-				}
-			} catch (ServerNotAvailableException $e) {
-				if (\trim($this->configuration->ldapBackupHost) === "") {
-					throw $e;
+		$backupHostEmpty = \trim($this->configuration->ldapBackupHost) === '';
+
+		try {
+			// skip contacting main server after failed connection attempt
+			// until cache TTL is reached
+			if ($backupHostEmpty
+				|| (!$this->configuration->ldapOverrideMainServer
+				&& !$this->getFromCache('overrideMainServer'))
+			) {
+				$this->doConnect(
+					$this->configuration->ldapHost,
+					$this->configuration->ldapPort
+				);
+				if (@$this->ldap->bind(
+					$this->ldapConnectionRes,
+					$this->configuration->ldapAgentName,
+					$this->configuration->ldapAgentPassword
+				)) {
+					$this->checkError('Bind success', ['app' => __METHOD__.':'.__LINE__]);
+					return true;
 				}
-			} catch (BindFailedException $e) {
-				if (\trim($this->configuration->ldapBackupHost) === "") {
-					throw $e;
+				$this->checkError('Bind failed', ['app' => __METHOD__.':'.__LINE__]);
```

Comment:
this line is not reachable  when bind is failed. In this case `OC\ServerNotAvailableException` exception is thrown and we   are falling into the catch clause (Line 563 below) where this exception is rethrown if there are no more connections